b1ddi/util/util.go

```diff
@@ -0,0 +1,61 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"runtime/debug"
+	"strconv"
+)
+
+var (
+	ParseError = "unable to parse key '%s': %v"
+	goSDKPath  = "github.com/infobloxopen/b1ddi-go-client"
+)
+
+func ToInt(d map[string]interface{}, key string) (int, bool, error) {
+	var (
+		i   int
+		err error
+		ok  bool
+		val interface{}
+	)
+	if val, ok = d[key]; ok {
+		i, err = strconv.Atoi(val.(string))
+		if err != nil {
+			err = errors.New(fmt.Sprintf(ParseError, key, err))
+		}
+	}
+	return i, ok, err
+}
+
+func ToBool(d map[string]interface{}, key string) (bool, bool, error) {
```

Comment:
Why do you need second bool (ok) return value, if you have an error, which you can always check for nil value?

Please simplify this
```suggestion
func ToBool(d map[string]interface{}, key string) (bool, error) {
```

Reply:
I need the boolean value because the key could be optional. 
Not all the keys in the DNS Record option and rData are required. Consider this scenario: if let's say the key doesn't exist, the error would be nil and w/o this boolean I would have a way to tell if the update was required or not in the calling function.